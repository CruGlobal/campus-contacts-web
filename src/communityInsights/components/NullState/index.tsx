import React from 'react';
import styled from '@emotion/styled';
import { withTheme } from 'emotion-theming';
import { useTranslation } from 'react-i18next';

interface Props {
    impactInfo?: boolean;
    width?: number;
}

const NullState = ({ impactInfo, width }: Props) => {
    const LoadingContainer = styled.div`
        width: 100%;
        height: ${width}px;
        background-color: ${({ theme }) => theme.colors.white};
        animation: fade 2s infinite linear both;
        border-radius: 8px;
        @keyframes fade {
            from {
                background-color: ${({ theme }) => theme.colors.white};
            }
            to {
                background-color: ${({ theme }) => theme.colors.highlight};
            }
        }
    `;
    const { t } = useTranslation('insights');

    return impactInfo ? (
        <span>{t('gatheringStats')}</span>
    ) : (
        <LoadingContainer />
    );
};

export default withTheme(NullState);
