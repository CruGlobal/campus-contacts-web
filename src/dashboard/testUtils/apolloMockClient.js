import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
