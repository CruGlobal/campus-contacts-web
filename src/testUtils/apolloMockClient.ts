import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
