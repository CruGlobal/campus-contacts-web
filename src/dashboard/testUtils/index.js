import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
