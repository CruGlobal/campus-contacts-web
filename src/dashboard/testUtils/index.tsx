import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
