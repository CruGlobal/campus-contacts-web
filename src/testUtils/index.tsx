import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
