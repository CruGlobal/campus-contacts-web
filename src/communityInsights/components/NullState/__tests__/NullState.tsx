import React from 'react';
import NullState from '..';
import { renderWithContext } from '../../../../testUtils';

describe('<NullState /> for non ImpactInfo component', () => {
    it('Should render properly', () => {
        const width = 400;
        const { snapshot, unmount } = renderWithContext(
            <NullState width={width} />,
        );
        snapshot();
        unmount();
    });
});

describe('<NullState /> for ImpactInfo', () => {
    it('Should render correct text', () => {
        const { snapshot, unmount } = renderWithContext(
            <NullState impactInfo={true} />,
        );
        snapshot();
        unmount();
    });
});
