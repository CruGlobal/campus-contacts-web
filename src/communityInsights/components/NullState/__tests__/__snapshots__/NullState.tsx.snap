// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NullState /> for non ImpactInfo component Should render properly 1`] = `
.emotion-2 {
  width: 100%;
  height: 190px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 8px;
}

.emotion-0 {
  text-align: center;
  width: 720px;
  color: #007398;
  font-size: 16px;
  font-weight: 300;
  line-height: 24px;
  margin: 0;
}

<div>
  <div
    class="emotion-2"
  >
    <img
      src="mockFile"
    />
    <p
      class="emotion-0"
    >
      This community does not have any members who have completed a personal step of faith. Challenge your
    </p>
    <p
      class="emotion-0"
    >
      community to open the MissionHub mobile app and complete a personal step of faith.
    </p>
  </div>
</div>
`;
